doc/api/esm.md

```diff
@@ -1024,6 +1024,7 @@ this one, or `undefined` if this is the main entry point for the application.
  * @param {string} specifier
  * @param {object} context
  * @param {string} context.parentURL
+ * @param {string[]} context.conditions
```

Comment:
can we add a note that passing `"require"` may be a breaking change in the future. If we end up causing it to route through the CJS resolver.

Reply:
Should we disallow passing `require` right now to allow for that future possibility? Allowing import-style resolution but with `require` condition seems bad™.

Reply:
I'm ok throwing for now, but that seems a bit perf heavy if you keep missing parse cache. I do think the resolve hook works fine for `require` design wise, but thats a separate discussion than this PR.

Reply:
Added an error that is thrown when a loader tries to add `require` as a condition.